hikari/utilities/enums.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+# cython: language_level=3
+# Copyright (c) 2020 Nekokatt
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Implementation of parts of Python's `enum` protocol to be faster."""
+from __future__ import annotations
+
+__all__: typing.List[str] = ["Enum"]
+
+import os
+import sys
+import types
+import typing
+
+_T = typing.TypeVar("_T")
+
+
+class _EnumNamespace(typing.Dict[str, typing.Any]):
+    __slots__: typing.Sequence[str] = ("base", "names_to_values", "values_to_names")
+
+    def __init__(self, base: typing.Type[typing.Any]) -> None:
+        super().__init__()
+        self.base = base
+        self.names_to_values: typing.Dict[str, typing.Any] = {}
+        self.values_to_names: typing.Dict[str, typing.Any] = {}
+        self["__doc__"] = "An enumeration."
```

Comment:
I'm not exactly sure how this works in the inheritance tree but just making sure that this doesn't override any set enums on the classes that inherit from this right?

Reply:
No, it doesnt affect it at the slightest. 
![image](https://user-images.githubusercontent.com/29100934/93916394-c4697500-fd09-11ea-8eec-488079a6b18d.png)

It will only show that doc if the subclass doesn't overwrite the doc